### ISSUE TITLE
Consider other tools

### ISSUE THREAD

**Issue by MaxUlysse** [nf-core/sarek#95]
Issue by @MaxUlysse, moved from SciLifeLab#666

- [ ] [ExpansionHunter](https://github.com/Illumina/ExpansionHunter) for estimating repeat sizes
- [ ] [QDNAseq](https://github.com/ccagc/QDNAseq) CNVs for shallow WGS
- [ ] [CNVkit](https://github.com/etal/cnvkit) for CNVs
- [ ] [deconstructSigs](https://github.com/raerose01/deconstructSigs) for mutational signatures
- [ ] ditto [mutation-signatures](https://github.com/mskcc/mutation-signatures) 
- [ ] [SomaticSeq](https://github.com/bioinform/somaticseq) ensemble caller and machine learning
- [ ] [MSIsensor](https://github.com/ding-lab/msisensor) for replication slippage variants at microsatellite regions
- [ ] [Oncotator](https://portals.broadinstitute.org/oncotator/) for annotating human point mutations and indels
- [ ] [MutSig](https://software.broadinstitute.org/cancer/cga/mutsig) for Mutation Significance checks
- [ ] [HsMetrics](https://broadinstitute.github.io/picard/javadoc/picard/picard/analysis/directed/HsMetrics.html) better QC 
- [ ] [CNNScoreVariants](https://software.broadinstitute.org/gatk/documentation/tooldocs/4.0.3.0/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNScoreVariants.php) Annotate a VCF with scores from a Convolutional Neural Network (CNN). 
- [ ] [SignatureAnalyzer](https://software.broadinstitute.org/cancer/cga/msp) yet an other mutation signature analyser from Broad
- [ ] [PROSIC2](https://github.com/prosic/prosic2) is a caller for somatic variants in tumor-normal sample pairs
- [ ] [seqCAT](https://github.com/fasterius/seqCAT) Evaluation of SNV profiles
- [ ] [varlociraptor](https://github.com/varlociraptor/varlociraptor) A Rust library for variant calling using a latent variable model
- [ ] [GATK CNN](https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_vqsr_CNNScoreVariants.php) Convolutional Neural Net to filter annotated variants

**Comment by davidmasp** [comment]
Would something like [somalier](https://github.com/brentp/somalier) be interesting? just saw in twitter that now it can be integrated in MultiQC. I am not sure if there is a conda for that yet though...

**Comment by MaxUlysse** [comment]
Nothing yet in Bioconda.
What would be your use case with somalier?

**Comment by davidmasp** [comment]
Maybe not superuseful in terms of new features. I understand it identifies that tumor/ normal come from the same patient. I think it could be useful in some set ups and maybe also somatic calling for cell lines for instance which aren’t exactly a classical use case for a somatic vc pipeline. Indeed idk about the tool much though.

**Comment by mvelinder** [comment]
I'd throw my support in for DeepVariant integration under `--tools`

Is that still a priority? It seemed like it was being worked on back in #105 

Thanks!

**Comment by mvelinder** [comment]
ExpansionHunter would also be a great addition under `--tools`

**Comment by maxulysse** [comment]
Hi @mvelinder both are tools that we indeed considered.
At the time of #105 deepvariant seemed to big to be integrated into sarek.
with DSL2, it would make sense to have it within sarek, so we currently have plan to have all the necessary modules made in nf-core/modules and then integrate into sarek.

**Comment by egenomics** [comment]
Hi,

For clinical environments it would be great to have a depth calculation tool. Mosdepth is a winner, as already suggested.

Another important tool would be a cnv caller. Manta is not a good tool for it, at least for gene panels. An easy and validated solution would be Exomedepth:
https://pubmed.ncbi.nlm.nih.gov/32561899/

**Comment by abhi18av** [comment]
I'd like to suggest addition of [`DeepTrio` ](https://github.com/google/deepvariant/blob/r1.1/docs/deeptrio-details.md)which builds upon `DeepVariant` is aimed at variant calling of trios or duos.

**Comment by FriederikeHanssen** [comment]
Expansionhunter & HSMetrics have both been suggested by @marchoeppner https://nfcore.slack.com/archives/CGFUX04HZ/p1647518666493429?thread_ts=1647517444.673109&cid=CGFUX04HZ something for 3.1?

**Comment by gorgitko** [comment]
Hi, `deepvariant` has been already incorporated, but without `deepTrio`, right?

**Comment by FriederikeHanssen** [comment]
yes that is right

**Comment by gorgitko** [comment]
Would it be possible to include it in TODO (board)? :innocent: Thanks!

**Comment by tdanhorn** [comment]
I second the inclusion of Picard Tools CollectHsMetrics (for WES/targeted sequencing), and there is CollectWgsMetrics for WGS.  Picard is already there, so all that is needed is an option to run it.  I find HsMetrics very useful for assessing coverage (e.g. it can tell you if deeper sequencing might solve a coverage problem or not), and while I can run it manually, it would be much nicer to just have it included.  In addition to the genome and the BAM, it needs intervals for targets (i.e. the existing `--target_bed` parameter) and for baits (would need an equivalent parameter for this).  These tools also work nicely with MultiQC.  Thank you!

**Comment by praveenraj2018** [comment]
It would be nice to have more SV callers in Sarek for better characterization, especially for tumor and normal samples. It is always a challenge to figure-out true/significant calls when there are complex rearrangements exist. 

_I suggest to include:_

**GRIDSS2** - Comprehensive characterization of SVs (https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02423-x)

and then,

**LINX**  - algorithm classifies somatic structural variation in tumors; It does chaining, clustering, and visualizations provide insights into complex rearrangements (https://www.sciencedirect.com/science/article/pii/S2666979X22000325)

**It has been demonstrated in several papers recently in oncology:**

https://www.nature.com/articles/s41698-021-00155-6
https://www.nature.com/articles/s41591-020-1072-4.pdf?proof=tr
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7545150/
https://www.biorxiv.org/content/10.1101/781013v1
https://www.frontiersin.org/articles/10.3389/fonc.2022.919118/full
https://pubmed.ncbi.nlm.nih.gov/35047820/

**Comment by mvelinder** [comment]
octopus also looked very promising 

https://github.com/luntergroup/octopus

https://www.nature.com/articles/s41587-021-00861-3

**Comment by maxulysse** [comment]
I was sure it was on our list already

**Comment by maxulysse** [comment]
@lbeltrame (cc @LMannarino @aldosr @rikizad) to add a tool I'd recommend to create a modules in the nf-core/modules repo following this documentation: https://nf-co.re/developers/modules#writing-a-new-module-reference
Please don't hesitate to ping me (either here or on Slack if you need any pointers).

**Comment by lbeltrame** [comment]
Thanks, Maxime. 
We have experience writing nf-core modules (we have a handful we use internally, ranging from local modifications to entire pipelines), but I'll make sure to ping if need be (tests can be... problematic ;).

I'll file separate issues on nf-core/modules to track this.

**Comment by maxulysse** [comment]
I look forward to the opportunity to collaborate

**Comment by nevinwu** [comment]
Hi all,
I'm currently working on a lofreq_somatic module to perform somatic variant calling on paired samples with the lofreq caller. Once it is finished, maybe it would be interesting adding it to sarek as another caller. Tumor-only modes of lofreq have already been implemented and are part of nf-core/modules (lofreq_call and lofreq_callparallel).

**Comment by Lumimar** [comment]
I am wondering whether there might be scope to extend sarek functionality to add:

- **Gem mapper** - https://www.nature.com/articles/nmeth.2221 ([review](https://www.nature.com/articles/nmeth.2256) states “The Genome Multitool (GEM) mapper rapidly and accurately provides all alignments of a read within a user-defined number of mismatches.”). Github: https://github.com/smarco/gem3-mapper

- **SiNPle** (https://www.mdpi.com/2073-4425/10/8/561) “SiNPle (Simplified Inference of Novel Polymorphisms from Large coveragE), a fast and effective software for variant calling. SiNPle is based on a simplified Bayesian approach to compute the posterior probability that a variant is not generated by sequencing errors or PCR artefacts.” Github: https://github.com/PaoloRibeca/SiNPle

Many thanks!

**Comment by maxulysse** [comment]
Consider adding Normalisation for VCFs before annotation

**Comment by SofiaMAhmed** [comment]
I would like Octopus to be added to Sarek  if possible

**Comment by maxulysse** [comment]
> I would like Octopus to be added to Sarek if possible

I considered that a while back, but the installation was not easy, it seems that it's available on bioconda nowadays, so I'll definitively have a look into it: https://bioconda.github.io/recipes/octopus/README.html

Do you know if the tool is still maintained, I don't see much activity on the github repo

**Comment by eolaniru** [comment]
would be nice to have  [vardictjava](https://github.com/AstraZeneca-NGS/VarDictJava) , a sensitive variant caller added to sarek

**Comment by bounlu** [comment]
Any **fusion** detection tool such as [LINX](https://github.com/hartwigmedical/hmftools/tree/master/linx) will be a great addition.

**Comment by sci-kai** [comment]
It would be nice to have [delly](https://github.com/dellytools/delly?tab=readme-ov-file#delly) integrated into sarek.

It is another SV caller, an [nf-core module](https://nf-co.re/modules/delly_call/) already exists so I think it might be easy to add this as another variant calling option to the pipeline.
Since manta is no longer updated, delly is a good addition that get regular updates and improvements.
It has a germline and somatic mode.

**Comment by j-gillespie-dna** [comment]
Not a tool but perhaps a way to download and create a cache for all resources? For example, I am running sarek and it would be great to cache all of the reference files the first time through so I could run offline in subsequent batches.

**Comment by georgiakes** [comment]
Hellooo . I think it would be nice to add `bcftools +fill-tags` in the post-variant calling options. Thank you :)
